tests/ui/field_reassign_with_default.rs

```diff
@@ -86,15 +86,6 @@ fn main() {
     c.0 = 42;
     c.1 = 21;
 
-    // wrong, produces the fifth error in stderr
-    let mut a: A = Default::default();
-    a.i = Default::default();
-
-    // wrong, produces the sixth error in stderr
-    let mut a: A = Default::default();
-    a.i = Default::default();
-    a.j = 45;
-
```

Comment:
These should still be linted. 

> * Fix assumption that `field: Default::default()` is the same as `..Default::default()`

I think removing this assumptions stops this from being linted? Can you leave those tests in and add a comment, that we can't lint this, because `Default` could be implemented by hand and therefore `a.i = default()` might be different from `..default()`?

Reply:
I will put the tests back. They are still linted and the default value is treated like any other value, not special-cased.